### PR TITLE
Add validation for ibc denom markers

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

Adding validation to markers if the denom starts with `ibc/`.

closes: #1289 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/marker/types/marker.go [hunk 1]
```diff
@@ -167,6 +172,23 @@ func (ma MarkerAccount) Validate() error {
 	return ma.BaseAccount.Validate()
 }
 
+func ValidateIbcDenom(ma MarkerAccount) error {
+	if !strings.HasPrefix(ma.Denom, "ibc/") {
+		return nil
+	}
+	if ma.SupplyFixed {
+		return errors.New("supply fixed is not supported for ibc marker")
```
[the remark this answers is not in the document]
Comment (nullpointer0x00) [reply x/marker/types/marker.go]: Good suggestion.  Changed.

File: cmd/provenanced/cmd/genaccounts.go [hunk 2]
```diff
@@ -420,7 +420,9 @@ enforced immediately.  An optional type flag can be provided or the default of C
 				managerAddr,
 				accessGrants,
 				markerStatus,
-				markertypes.MarkerType(markerType))
+				markertypes.MarkerType(markerType),
+				true,
```
Comment (egaxhaj) [inline cmd/provenanced/cmd/genaccounts.go:424]: Is it always a fixed supply?

Comment (nullpointer0x00) [reply cmd/provenanced/cmd/genaccounts.go:424]: For this it currently is.  If you look in the marker types, I changed a constructor `NewMarkerAccount` to take in a fixed supply parameter that was automatically set to true before.  I didn't want to change the behavior for anything that was out of scope for the ticket with this minor refactor.  So, I just hard coded the true here.  However, if you look in the msg_server.  We have a few instances of using `NewMarkerAccount` then the next line would change the supply fixed value.  Now that is fixed to just take it in as a param.

**Review by egaxhaj — APPROVED** [review]
LGTM

**Pull request merged**